packages/core/src/scp-cf/destination-accessor.ts

```diff
@@ -482,19 +285,346 @@ const emptyDestinationByType: DestinationsByType = {
   subaccount: []
 };
 
-function subscriberDestinationIsSelected(
-  selectionStrategy: DestinationSelectionStrategy
-): boolean {
-  return (
-    selectionStrategy === subscriberFirst ||
-    selectionStrategy === alwaysSubscriber
-  );
-}
-
 function emptyDestinationsByType(
   destinationByType: DestinationsByType
 ): boolean {
   return (
-    !destinationByType.instance.length && !destinationByType.instance.length
+    !destinationByType.instance.length && !destinationByType.subaccount.length
   );
 }
+
+class DestinationAccessor {
+  public static async getDestinationFromDestinationService(
+    name: string,
+    options: DestinationOptions & { iss?: string }
+  ): Promise<Destination | undefined> {
```

Comment:
I tried will use `null` here but keep the `undefined` everywhere else because it is more convenient because you do not need explicit return statements.